### ISSUE TITLE
Finish remaining work needed to announce Atom Nightly channel

### ISSUE THREAD

**Issue by daviwil** [atom/atom#17884]
This issue tracks the remaining tasks that need to be completed before we can fully announce the Atom Nightly channel for general use:

- [x] Enable publishing of Atom Nightly packages to packagecloud (add `--linux-repo-name "atom-staging"` to [this line](https://github.com/atom/atom/blob/50d890abea8d66302a458c599b11951b4c8e6860/script/vsts/nightly-release.yml#L57) in `nightly-releases.yml`
- [x] [Enable side-by-side Atom channel installs on Windows](https://github.com/atom/atom/pull/17813) -- Not entirely necessary, but very nice to have
- [x] [Fix issue preventing Nightly from updating past `-nightly9` on Windows](https://github.com/atom/atom/issues/18417) (see https://github.com/atom/atom/issues/18417#issuecomment-492435520)
- [x] [Enable 32-bit builds of Nightly channel](https://github.com/atom/atom/issues/17668)
- [x] Update and merge new Atom Nightly landing page (https://github.com/github/atom.io/pull/1261)
- [x] Publish blog post (https://github.com/atom/blog.atom.io/pull/236) -- make sure to update the "Caveats" section to account for anything that has been improved since the post was originally written

**Comment by rsese** [comment]
Should we add https://github.com/atom/atom/issues/18417 to the list @daviwil?

**Comment by rafeca** [comment]
Here goes some guesstimation about the remaining work that needs to be done for having nightly releases.

My current estimate is **~1.5 weeks** of actual work (4 days for the actual tasks and 4 more days to do proper testing/verification of the whole pipeline + fixing potential issues).

This is the actual time that I estimate we need to spend working on it, but since we have external dependencies that we can't easily control, shipping this will probably take longer than 1.5 weeks (due to the time we'll be blocked).

## Required work

### 1. Enable publishing of linux Atom Nightly packages to packagecloud.

**Very Small:** it can be done in <30min.

### 2. Tweak [the announcement blog post](https://github.com/atom/blog.atom.io/pull/236)

**Small:** it can be done in ~1h.

We need to update it with remaining suggestions and update the caveats.

### 3. Add functionality to the [Atom.io landing page PR](https://github.com/github/atom.io/pull/1261)

**Small-Medium:** it can be done in half a day.

### 4. Fix `Squirrel.Windows` to support upgrading to versions above `nightly-9`

**Medium, but potentially large:** I estimate the work we need to do to take 1-3 days, but we may need to wait for some time until the fixes propagate to Atom.

There are a few things that need to be done. Some of them don't depend on us:

* Upgrade Atom's `electron-winstaler` from v2.6.4 to v3.0.4. This may not be trivial since [we're using a 2 year old version](https://github.com/electron/windows-installer/compare/v2.6.4...v3.0.4). I've created [a WIP PR](https://github.com/atom/atom/pull/19539).
* Wait until https://github.com/Squirrel/Squirrel.Windows/pull/1492 gets merged.
* Wait until a new version of `Squirrel.Windows` gets published.
* Wait until `electron-winstaler` upgrades to the new version of `Squirrel.Windows` and a new version is published.
* Upgrade Atom's `electron-winstaler` to published version.

## Optional work

### 5. Enable side-by-side Atom channel installs on Windows 

**Medium:** it can take 1-2 days. Most of the work has been already done, only some minor work is required on atom.io ([comment](https://github.com/atom/atom/pull/17813#issuecomment-492424136)), but this will require proper testing.

* Update download/update logic on atom.io. Looks like we need to change [these 2 RegExps](https://github.com/github/atom.io/blob/47352821b2f9b6ee3300983b7ca32899cadc12a4/app/models/release_index.rb#L20-L22).
 * ~~Update the `atom-release-publisher` to upload the new filenames~~ (we don't need to do anything since [it already handles dynamic `nupkg` file names](https://github.com/atom/atom-release-publisher/blob/652df69ed7a4096cf3073ccfd4a47ba5d9b9934b/src/artifact-downloader.js#L51)).
  * ~~Update the draft release publisher to handle new filenames~~ (we don't need to do anything since [it already handles dynamic `nupkg` file names](https://github.com/atom/atom/blob/master/script/vsts/upload-artifacts.js#L41)).

This is nice to have, but it's not a requisite for unblocking nightly releases.

**Comment by rafeca** [comment]
Good news everyone! we've been able to solve the 4th point of the required steps (the most complex one) by publishing a custom version of `electron-winstaller` called `@atom/electron-winstaller` which contains the fixes made in https://github.com/Squirrel/Squirrel.Windows/pull/1492.

With this, the rest of needed work should be trivial 😄

**Comment by rafeca** [comment]
I've enabled nightlies on Linux, updated and tweaked the blog post PR and the `atom.io` nighly page PR, with this we're ready to ship nightlies!

Unless somebody thinks it's a bad idea, my current plan is to ship the blog post and the nighly page on atom.io tomorrow **June 21st**.

**Comment by Aerijo** [comment]
The Linux nightlies on Ubuntu through `apt` seem to conflict with beta
```
Unpacking atom-nightly (1.40.0-nightly12) over (1.36.0-nightly7) ...
dpkg: error processing archive /var/cache/apt/archives/atom-nightly_1.40.0-nightly12_amd64.deb (--unpack):
 trying to overwrite '/usr/share/polkit-1/actions/atom.policy', which is also in package atom-beta 1.39.0-beta2
Errors were encountered while processing:
 /var/cache/apt/archives/atom-nightly_1.40.0-nightly12_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

**Comment by rafeca** [comment]
urggg! 😟 thanks for the heads up @Aerijo !

@as-cii that issue is related to https://github.com/atom/atom/pull/19412 and will get propagated to stable on the next release.

We should probably have different policy files depending on the release channel, e.g `atom-stable.policy`, `atom-beta.policy` and `atom-nightly.policy`.
